src/Symfony/Component/FeatureFlag/FeatureCheckerInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag;
+
+interface FeatureCheckerInterface
+{
+    /**
+     * @param string $featureName   the name of the feature to check
+     * @param mixed  $expectedValue comparison value required to determine if the feature is enabled
+     */
+    public function isEnabled(string $featureName, mixed $expectedValue = true): bool;
```

Comment:
I don't think that it brings a lot of complexity since only the value is cached, not the comparison.

Reply:
I find that having this extra argument blurs the goal of the method for newcomers and creates a fixed API that it will be hard to get away from. I suggest that if this is indeed useful, to add it in a later version. Also what the cache currently is and what it could become with custom implems are not the same thing. By exposing this argument devs may take it into account for caching and thus creating complexity to extend / decorate this.

Reply:
>  By exposing this argument devs may take it into account for caching and thus creating complexity to extend / decorate this.

Devs may have to take it into account... or not 🙂
Also, this kind of cache could be done in the provider because caching should only concern the value of a feature, not if its enabled or not. 

IMO, this second argument is useful to make the checker's usage easier:
```php
$this->featureChecker->isEnabled('version', 2);
// vs.
2 === $this->featureChecker->getValue('version');
```

Reply:
I don't find it easier and what is the goal to have a parameter devs would not take it into account ? I don't get it. At best I find it harder to read withe the second parameter as it loks like a fallback value...